lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -384,6 +398,34 @@ data ApiAddress (n :: NetworkDiscriminant) = ApiAddress
     , state :: !(ApiT AddressState)
     } deriving (Eq, Generic, Show)
 
+newtype ApiScript = ApiScript
+    { script :: ApiT Script
+    } deriving (Eq, Generic, Show)
+
+newtype ApiPubKey = ApiPubKey
+    { pubKey :: ByteString
+    } deriving (Eq, Generic, Show)
+
+data Credential = Credential
+    { credential :: Either ApiPubKey ApiScript
```

Comment:
May I suggest a sum-type instead of wrapping a `Either` ? 

```hs
data Credential 
    = CredentialPubKey ...
    | CredentialScript
```

?

Reply:
Should it be called ApiCredential as well for consistency?

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/2253/commits/1cdaf25774c19ac6a3b37e0cebb48b9628095f19